hr_course/models/hr_course.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrCourse(models.Model):
+    _name = 'hr.course'
+    _inherit = 'mail.thread'
+
+    name = fields.Char(
+        string='Name', required=True,
+        track_visibility="onchange"
+    )
+    category_id = fields.Many2one(
+        'hr.course.category',
+        string='Category',
+        required=True,
+    )
+
+    start_date = fields.Date(
+        string="Start date", readonly=True,
+        states={'draft': [('readonly', False)]},
+        track_visibility="onchange",
+    )
+    end_date = fields.Date(
+        string="End date", readonly=True,
+        states={'draft': [('readonly', False)]},
+        track_visibility="onchange",
+    )
+    currency_id = fields.Many2one(
+        'res.currency', string='Currency',
+        default=lambda self: self.env.user.company_id.currency_id,
+    )
+    cost = fields.Monetary(
+        string='Course Cost', required=True,
+        track_visibility="onchange"
+    )
+    authorized_by = fields.Many2one(
+        string="Authorized by",
+        comodel_name='hr.employee',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        track_visibility="onchange"
+    )
+    permanence = fields.Boolean(
+        string="Has Permanence",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        track_visibility="onchange"
+    )
+    permanence_time = fields.Char(
+        string='Permanence time',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        track_visibility="onchange"
+    )
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('waiting_attendees', 'Waiting attendees'),
+            ('in_progress', 'In progress'),
+            ('in_validation', 'In validation'),
+            ('completed', 'Completed'),
+            ('cancelled', 'Cancelled'),
+        ],
+        required=True,
+        readonly=True,
+        default='draft',
+        track_visibility="onchange"
+    )
+    attendant_ids = fields.Many2many(
+        'hr.employee',
+        readonly=True,
+        states={'waiting_attendees': [('readonly', False)]},
+    )
+    course_attendee_ids = fields.One2many(
+        'hr.course.attendee',
+        inverse_name='course_id',
+        readonly=True,
+        states={'in_validation': [('readonly', False)]},
+    )
+
+    @api.constrains('start_date', 'end_date')
+    def _check_start_end_dates(self):
+        self.ensure_one()
+        if self.start_date and self.end_date:
```

Comment:
@jarroyomorales `if self.start_date and self.end_date and self.start_date > self.end_date:`